src/rust/lib/frp/src/io/keyboard.rs

```diff
@@ -47,74 +47,49 @@ impl KeyMask {
     }
 
     /// Check if key bit is on.
-    pub fn has_key(&self, key:&Key) -> bool {
+    pub fn contains(&self, key:&Key) -> bool {
         let KeyMask(bit_set) = self;
-        bit_set.get_bit(Self::bit_position_for(key))
+        bit_set.get_bit(key.legacy_keycode() as usize)
     }
 
     /// Set the `key` bit for new state.
-    pub fn set_key(&mut self, key:&Key, state:bool) {
+    pub fn set(&mut self, key:&Key, state:bool) {
         let KeyMask(ref mut bit_set) = self;
-        bit_set.set_bit(Self::bit_position_for(key),state);
-    }
-}
-
-impl Default for KeyMask {
-    fn default() -> Self {
-        let mut bitset = DenseBitSetExtended::with_capacity(MAX_KEY_CODE + 1);
-        // This is the only way to set bitset length.
-        bitset.set_bit(MAX_KEY_CODE,true);
-        bitset.set_bit(MAX_KEY_CODE,false);
-        Self(bitset)
+        bit_set.set_bit(key.legacy_keycode() as usize,state);
     }
 }
 
 impl<'a> FromIterator<&'a Key> for KeyMask {
     fn from_iter<T: IntoIterator<Item=&'a Key>>(iter:T) -> Self {
         let mut key_mask = KeyMask::default();
         for key in iter {
-            let bit = Self::bit_position_for(key);
+            let bit = key.legacy_keycode() as usize;
             key_mask.set_bit(bit,true);
         }
         key_mask
     }
 }
 
-impl From<&[Key]> for KeyMask {
-    fn from(keys: &[Key]) -> Self {
-        <KeyMask as FromIterator<&Key>>::from_iter(keys)
-    }
-}
+impl From<&[Key]>   for KeyMask { fn from(keys:&[Key])   -> Self { KeyMask::from_iter(keys) } }
+impl From<&[Key;1]> for KeyMask { fn from(keys:&[Key;1]) -> Self { KeyMask::from_iter(keys) } }
+impl From<&[Key;2]> for KeyMask { fn from(keys:&[Key;2]) -> Self { KeyMask::from_iter(keys) } }
+impl From<&[Key;3]> for KeyMask { fn from(keys:&[Key;3]) -> Self { KeyMask::from_iter(keys) } }
+impl From<&[Key;4]> for KeyMask { fn from(keys:&[Key;4]) -> Self { KeyMask::from_iter(keys) } }
+impl From<&[Key;5]> for KeyMask { fn from(keys:&[Key;5]) -> Self { KeyMask::from_iter(keys) } }
 
 
-// === Private ===
-
-impl KeyMask {
-    fn bit_position_for(key:&Key) -> usize {
-        (match key {
-            // On Chrome, shift+alt gives `Meta` key, therefore we do a unification here.
-            Key::Meta => Key::Alt.legacy_keycode(),
```

Comment:
What about this reason I written here? Have you checked that Shift+Alt operations are working properly? (For example Alt + Shift + Mouse click)

Reply:
Yes, alt + shift returns a valid key map on my side in chrome here.